tests/Unit/Domain/Test_ElementDistribution.cpp

```diff
@@ -3,306 +3,780 @@
+#include "Helpers/Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
 
 namespace {
-
-// a global indexing to help aggregating the processor maps in the tests
+// Make a `domain::creators::AlignedLattice` from an option string
 template <size_t Dim>
-std::array<SegmentId, Dim> element_index_to_segment_id(
-    const std::array<size_t, Dim>& refinement_levels,
-    const size_t element_index) {
-  std::array<SegmentId, Dim> segment_ids{};
-  size_t stride = 1;
-  for (size_t i = 0; i < Dim; ++i) {
-    const size_t element_index_in_dim =
-        (element_index / stride) % two_to_the(gsl::at(refinement_levels, i));
-    segment_ids.at(i) =
-        SegmentId{gsl::at(refinement_levels, i), element_index_in_dim};
-    stride *= two_to_the(gsl::at(refinement_levels, i));
-  }
-  return segment_ids;
+auto make_domain_creator(const std::string& opt_string) {
+  return TestHelpers::test_option_tag<
+      domain::OptionTags::DomainCreator<Dim>,
+      TestHelpers::domain::BoundaryConditions::
+          MetavariablesWithoutBoundaryConditions<
+              Dim, domain::creators::AlignedLattice<Dim>>>(opt_string);
 }
 
-template <size_t Dim>
-size_t segment_id_to_element_index(
-    const std::array<SegmentId, Dim>& segment_ids) {
-  size_t element_index = 0;
-  size_t stride = 1;
-  for (size_t i = 0; i < Dim; ++i) {
-    element_index += stride * gsl::at(segment_ids, i).index();
-    stride *= two_to_the(gsl::at(segment_ids, i).refinement_level());
+// Metavariables for a `domain::creators::BinaryCompactObject`
+template <size_t Dim, bool EnableTimeDependentMaps, bool WithBoundaryConditions>
+struct Metavariables {
+  struct domain : tt::ConformsTo<::domain::protocols::Metavariables> {
+    static constexpr bool enable_time_dependent_maps = EnableTimeDependentMaps;
+  };
+  using system = tmpl::conditional_t<WithBoundaryConditions,
+                                     TestHelpers::domain::BoundaryConditions::
+                                         SystemWithBoundaryConditions<Dim>,
+                                     TestHelpers::domain::BoundaryConditions::
+                                         SystemWithoutBoundaryConditions<Dim>>;
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes = tmpl::map<tmpl::pair<
+        DomainCreator<3>, tmpl::list<::domain::creators::BinaryCompactObject>>>;
+  };
+};
+
+// Stringify a bool
+std::string stringize(const bool t) { return t ? "true" : "false"; }
+
+// Create an option string for a `domain::creators::BinaryCompactObject`
+std::string create_option_string(const bool excise_A, const bool excise_B,
+                                 const bool add_time_dependence,
+                                 const bool use_logarithmic_map_AB,
+                                 const bool use_equiangular_map,
```

Comment:
Please remove everything from this test that isn't relevant for the element distribution algorithm. For instance, you don't need to create the domain creator from options, just use the constructor. Boundary conditions aren't relevant either, so just delete everything related to boundary conditions. Tests that do too many things can be quite hard to maintain. Please test only the logic in `ElementDistribution.*pp`, no more and no less.